sql/parser/builtins.go

```diff
@@ -319,6 +322,99 @@ var builtins = map[string][]builtin{
 		},
 	},
 
+	// Timestamp/Date functions.
+
+	"age": {
+		builtin{
+			types:      typeList{timestampType},
+			returnType: DummyInterval,
+			impure:     true,
+			fn: func(args DTuple) (Datum, error) {
+				return DInterval{Duration: time.Now().Sub(args[0].(DTimestamp).Time)}, nil
+			},
+		},
+		builtin{
+			types:      typeList{timestampType, timestampType},
+			returnType: DummyInterval,
+			fn: func(args DTuple) (Datum, error) {
+				return DInterval{Duration: args[0].(DTimestamp).Sub(args[1].(DTimestamp).Time)}, nil
+			},
+		},
+	},
+
+	"current_date": {
+		builtin{
+			types:      typeList{},
+			returnType: DummyDate,
+			impure:     true,
+			fn: func(args DTuple) (Datum, error) {
+				return DDate{Time: time.Now().Truncate(24 * time.Hour)}, nil
+			},
+		},
+	},
+
+	"current_timestamp": {nowImpl},
+	"now":               {nowImpl},
+
+	"extract": {
+		builtin{
+			types:      typeList{stringType, timestampType},
+			returnType: DummyInt,
+			fn: func(args DTuple) (Datum, error) {
+				// extract timeSpan fromTime.
+				fromTime := args[1].(DTimestamp)
+				timeSpan := strings.ToLower(string(args[0].(DString)))
+				switch timeSpan {
+				case "year":
+					return DInt(fromTime.Year()), nil
+
+				case "quarter":
+					return DInt(fromTime.Month()/4 + 1), nil
+
+				case "month":
+					return DInt(fromTime.Month()), nil
+
+				case "week":
+					_, week := fromTime.ISOWeek()
+					return DInt(week), nil
+
+				case "day":
+					return DInt(fromTime.Day()), nil
+
+				case "dayofweek":
+					return DInt(fromTime.Weekday()), nil
+
+				case "dayofyear":
+					return DInt(fromTime.YearDay()), nil
+
+				case "hour":
+					return DInt(fromTime.Hour()), nil
+
+				case "minute":
+					return DInt(fromTime.Minute()), nil
+
+				case "second":
+					return DInt(fromTime.Second()), nil
+
+				case "millisecond":
+					return DInt(fromTime.Nanosecond() / 1000000), nil
```

Comment:
Use the constants from the time module for these conversions: `fromTime.Nanosecond() / time.Millisecond` (with casts as needed)